api/src/database/run-ast.ts

```diff
@@ -469,7 +469,7 @@ function mergeWithParentItems(
 
 			if (nestedNode.query.page && nestedNode.query.page > 1) {
 				parentItem[nestedNode.fieldKey] = parentItem[nestedNode.fieldKey].slice(
-					(nestedNode.query.limit ?? 100) * (nestedNode.query.page - 1)
+					(nestedNode.query.limit ?? Number(env.DEFAULT_QUERY_LIMIT)) * (nestedNode.query.page - 1)
```

Comment:
Should we add `DEFAULT_QUERY_LIMIT` and `MAX_QUERY_LIMIT` as numbers into the env typemap to reduce the need to cast them as `Number()`? 

https://github.com/directus/directus/blob/1bfd46044928a8192fbef7c1a507899b31642abf/api/src/env.ts#L274-L292

I assume this will help https://github.com/directus/directus/issues/9521#issuecomment-962502798 a little when that issue is being worked 👍

---

though understandably we're technically also not enforcing type for `STORAGE_LOCATIONS` for example, since it seems like we're doing `toArray(env.STORAGE_LOCATIONS)` whenever it's being used:

https://github.com/directus/directus/blob/1bfd46044928a8192fbef7c1a507899b31642abf/api/src/controllers/files.ts#L48

so perhaps those needs to be addressed in a separate PR.

Reply:
No, lets keep casting it where it's being used. I want to get rid of the top-level casting (https://github.com/directus/directus/issues/9521)

Reply:
> No, lets keep casting it where it's being used. I want to get rid of the top-level casting

Just to confirm I understand this correctly, this would mean we wouldn't have the list of hardcoded environment variables to type cast anymore? (as apposed to what was suggested in https://github.com/directus/directus/issues/9521#issuecomment-962502798, but understandably it can be outdated 👍)

Or did you perhaps meant getting rid of _auto_ type casting, not getting rid of those existing enforced type casting?

Reply:
Good question.. I think it might be easier to maintain over time if we expect every environment variable to be a strong coming out of env.ts. That also matches how env vars work in general 🤔

Reply:
> easier to maintain over time if we expect every environment variable to be a strong coming out of env.ts

Strongly agree with this. 👍 Should we proceed with this refactor for the env vars in this PR?

Reply:
Lets do it for the new env vars added in this PR, but lets keep the refactoring of the other env vars for another PR 👍🏻